frontend/util.lua

```diff
@@ -829,6 +833,26 @@ function util.removeFile(file)
     end
 end
 
+function util.writeToSysfs(val, file)
+    -- NOTE: We do things by hand via ffi, because io.write uses fwrite,
+    --       which isn't a great fit for procfs/sysfs (e.g., we lose failure cases like EBUSY,
+    --       as it only reports failures to write to the *stream*, not to the disk/file!).
+    local fd = C.open(file, bit.bor(C.O_WRONLY, C.O_CLOEXEC)) -- procfs/sysfs, we shouldn't need O_TRUNC
```

Comment:
We also dont have the hall_file path on any legacy kindles so should not be an issue in practice

Reply:
Oh, duh' ;o).

Reply:
Can you help with the tests? `device_spec` is failing, and i am not sure how to convert its expecting us to use os.execute when we don't use it anymore. line 216

Reply:
Is the "fuck it" approach enough to appease it?

```diff
diff --git a/spec/unit/device_spec.lua b/spec/unit/device_spec.lua
index 7302ee4f5..7e5abb025 100644
--- a/spec/unit/device_spec.lua
+++ b/spec/unit/device_spec.lua
@@ -40,7 +40,6 @@ describe("device module", function()
     after_each(function()
         mock_input.open:revert()
         os.getenv:revert()
-        os.execute:revert()
 
         os.getenv = osgetenv
         io.open = iopen
@@ -213,8 +212,6 @@ describe("device module", function()
             assert.is.same(kindle_dev.powerd.fl_intensity, 5)
 
             kindle_dev.powerd:toggleFrontlight()
-            assert.stub(os.execute).was_called_with(
-                "printf '%s' 0 > /sys/class/backlight/max77696-bl/brightness")
             -- Here be shenanigans: we don't override powerd's fl_intensity when we turn the light off,
             -- so that we can properly turn it back on at the previous intensity ;)
             assert.is.same(kindle_dev.powerd.fl_intensity, 5)
```

(It's not like we could stub the new FFI call anyway).

Reply:
That was my plan. i just wanted to see if you had a way to salvage the wreckage :)

Reply:
Besides going full tilt overengineering and stubbing `util.writeToSys` altogether (which would barely make sense), nope; so you've got my blessing ^^.